pyrobosim/pyrobosim/core/ros_interface.py

```diff
@@ -11,7 +13,8 @@
 
 class WorldROSWrapper(Node):
     """ ROS2 wrapper node for pyrobosim worlds. """
-    def __init__(self, world=None, name="pyrobosim", state_pub_rate=0.1):
+    def __init__(self, world=None, name="pyrobosim",
+                 num_threads=4, state_pub_rate=0.1):
```

Comment:
4 threads ought to be plenty but just an FYI, I've heard from Tyler that there's a very low cost involved with increasing `num_threads`. Setting it to match the num_threads available on your computer would be a good way to do it.

Reply:
Sure :)

```
sebastian@scastro-thinkpad:~$ python3
Python 3.10.6 (main, Nov  2 2022, 18:53:38) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.cpu_count()
8
```